File: base/protocol/FloatItem.php
<?php

namespace ffan\dop\protocol;

use ffan\dop\Exception;

/**
 * Class ItemFloat
 * @package ffan\dop\protocol
 */
class FloatItem extends Item
{
    /**
     * @var int 类型
     */
    protected $type = ItemType::FLOAT;

    /**
     * 设置默认值
     * @param string $value
     * @throws Exception
     */
    public function setDefault($value)
    {
        if (!preg_match('/^(-?\d+)(\.\d+)?$/', $value)) {
            throw new Exception($value . ' 不能作为 float 默认值');
        }
        $this->default = $value;
    }
}

File: test/java/DopTest/GsonTest/src/com/ffan/dop/FloatItem.java
package com.ffan.dop;

class FloatItem extends Item {
    /**
     * 值
     */
    private float value;

    /**
     * 构造函数
     */
    FloatItem(float value) {
        this.value = value;
        this.type = ItemType.FLOAT_TYPE;
    }

    /**
     * 获取值 String
     */
    public String getValueString() {
        return String.valueOf(this.value);
    }

    /**
     * 获取值 int
     */
    public long getValueInt() {
        return Math.round(this.value);
    }

    /**
     * 获取值double
     */
    public double getValueDouble() {
        return this.value;
    }

    /**
     * 获取值 float
     */
    public float getValueFloat() {
        return this.value;
    }

    /**
     * 获取值 bool
     */
    public boolean getValueBool() {
        return 0 != this.getValueInt();
    }
}